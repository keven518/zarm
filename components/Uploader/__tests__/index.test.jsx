import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Uploader from '../index';

describe('Uploader', () => {
  it('renders correctly', () => {
    const props = {
      accept: 'image/jpg, image/jpeg, image/gif, image/png',
      onChange: jest.fn(),
      children: 'foo',
    };
    const wrapper = render(<Uploader {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('select file', () => {
  //   const props = {
  //     accept: 'image/jpg, image/jpeg, image/gif, image/png',
  //     onChange: jest.fn(),
  //     onBeforeSelect: jest.fn(),
  //     children: <button>+</button>,
  //   };
  //   const wrapper = shallow(<Uploader {...props} />);
  //   wrapper.find('button').simulate('click');
  //   expect(props.onBeforeSelect).toBeCalled();
  // });
});
