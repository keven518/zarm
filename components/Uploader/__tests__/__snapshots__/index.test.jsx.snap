// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploader renders correctly 1`] = `
<div
  class="za-uploader"
>
  <input
    accept="image/jpg, image/jpeg, image/gif, image/png"
    class="za-uploader-input"
    type="file"
  />
  <span
    class="za-uploader-trigger"
  />
  foo
</div>
`;
